New Resources, Full Json Implementation for Activities

wchen-oxy opened the pull request:
-Updated format for each entry in Nutrition/Exercise
-Updated each Json Read function with code to catch Null exceptions
-Passed Data successfully to detailed activity using parceable
-Added basic detailed activity layout
-Added new image resources for icon and main activities

wchen-oxy committed:
Merge pull request #2 from oxy-compsci/master

Pulling a new copy

wchen-oxy committed:
Merge pull request #3 from oxy-compsci/master

Merging Most Up to Date Code

wchen-oxy committed:
Fixed Exceptions/Nulls

-Debugged Json file 
-covered null cases

wchen-oxy committed:
Added Resources/Fixed Layouts

wchen-oxy commented:
I am aware that Dan's and my item format have similar changes. I had already made my changes when he pull requested. Not sure what to do but I'll pull request for now since the bulk of my changes are distinct from his. @justinnhli @crzeller11

wchen-oxy commented:
What should I do for the failed Travis CI build? It says my png resource failed because it doesn't contain the signature png format. Why should that matter if the file itself ends in .png?

justinnhli commented:
1. What's the difference between this PR and #13?

2. It might be that it ends in .png but is actually a jpeg file.

wchen-oxy commented:
#13 has only itemformat changed for nutrition. I have that changed as well using different variables. However, everything else I have listed under the updates are different than #13.

So to be more specific:
Same:
-Updated format for each entry in Nutrition

Different
-Updated format for each entry in Exercise
-Updated each Json Read function with code to catch Null exceptions
-Passed Data successfully to detailed activity using parceable
-Added basic detailed activity layout
-Added new image resources for icon and main activities

I don't think that its a jpeg file. I exported it as a  png through photoshop. The metadata says it's a png as well. I could reupload it again though.

blackjamx5 commented:
I looked at the metadata for the pngs I uploaded and they don't appear to be jpegs

justinnhli commented:
It's a JPEG:

![screen shot 2018-04-15 at 14 36 00](https://user-images.githubusercontent.com/624737/38783685-70b2e742-40ba-11e8-92a9-0532ed14ea6c.png)

Open it in Preview/Photoshop or whatever and force save it as a png.

wchen-oxy committed:
Changed png image

justinnhli commented:
Re: the differences from #13 - I think you and @Danhuth need to figure out what the right thing to do is. I believe the changes currently conflict, which means at least parts of one of these PRs will be overwritten. Figure that out and tag me again.

wchen-oxy commented:
Sorry my mistake. I checked using preview but not using the detailed info. I've uploaded the new png and will figure out the changes with Dan

wchen-oxy committed:
Shortened Json code

wchen-oxy committed:
Reformatted Code

wchen-oxy commented:
I've talked it over with Dan and we're going to use my code. I updated my variable names so that they fit his documentation. I've also updated the code with Justin's suggestions to Dan.
@justinnhli @crzeller11 

The code should be ready for merging.

justinnhli commented on app/build.gradle:
```diff
@@ -49,6 +49,9 @@ dependencies {
     implementation 'com.google.android.gms:play-services-location:11.8.0'
 
+    //begin picasso
+    compile 'com.squareup.picasso:picasso:2.5.2'
```
use `implementation` instead of `compile`

crzeller11 replied:
^^^avoid so many warnings with gradle build

justinnhli replied:
This has yet to be fixed.

wchen-oxy replied:
Gosh, yes you're right. I didn't commit the gradle file since we established awhile ago to not upload builds generated by android studio. My mistake.

justinnhli commented on app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java:
```diff
@@ -0,0 +1,54 @@
+        Bundle data = getIntent().getExtras();
+        //extra.putSerializable("Exercise_List", (exerciseList));
+        //ExerciseItemFormat ExercistItem = (ExerciseItemFormat) data.getParcelable("ExerciseInfo");
+        //Bundle data = getIntent().getExtras();
```
delete unused code

justinnhli commented on app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java:
```diff
@@ -0,0 +1,54 @@
+import android.support.v7.app.AppCompatActivity;
+import android.widget.TextView;
+
+//Please ignore this page for now, this is a dummy page
```
comment no longer accurage

justinnhli commented on app/src/main/java/practicum/getfitla_v3/ExerciseDetailActivity.java:
```diff
@@ -0,0 +1,54 @@
+    private int passedid;
+    private String passedtitle;
+    private String passedshortdesc;
+    private String passedisboolean;
```
what is this variable?

justinnhli commented on app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:
```diff
@@ -1,46 +1,130 @@
+                               //String rating,
+                               // String price,
+                               int image, String preptime, String servingSize, String calories,
+                               String equipment, String process, String ingredients) {
```
reformat this line and remove the unused parameters

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Exercise.java:
```diff
@@ -93,14 +141,23 @@ public void onResult(JSONObject object) {
+
+        ExerciseItemFormat current = exerciseList.get(position);
+        Intent intent = new Intent(this, ExerciseDetailActivity.class);
+        //  ArrayList<ExerciseItemFormat> passedExercise = exerciseList;
```
delete

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,210 @@
+                System.out.println(nutritionList);
+
+            }
+        }).execute(EXERCISE_SHEET_URL);
```
EXERCISE_SHEET_URL in Nutrition.java?

crzeller11 replied:
Yes, also confused by this.

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,210 @@
-        recyclerView.setAdapter(mAdapter);
-        //Bind the custom onClickListener
-        mAdapter.setClickListener(this);
+        final Nutrition ThisNutrition = this;
```
variable naming

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,210 @@
+                        shortdesc,
+                        image,
+                        // rating,
+                        //   price,
```
delete

wchen-oxy replied:
I blanked out the things associated with rating and price because I had yet to add rating and prices to the google sheets and layouts. Can I keep it for now?

justinnhli replied:
I would add it to the Google Sheet as empty columns, then uncomment them here.

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Exercise.java:
```diff
@@ -13,55 +14,102 @@
+                int counter = 0;
+
+                try {
+                    name = row.getJSONArray("c").getJSONObject(counter).optString("v");
```
This gets repeated a lot - can you put it in a function instead?

justinnhli commented on app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java:
```diff
@@ -39,25 +36,28 @@ public itemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-        holder.textViewRating.setText(String.valueOf(exercise.getRating()));
-        holder.textViewPrice.setText(String.valueOf(exercise.getPrice()));
+        //holder.textViewRating.setText(String.valueOf(exercise.getRating()));
+        //holder.textViewPrice.setText(String.valueOf(exercise.getPrice()));
```
delete

justinnhli commented on app/src/main/java/practicum/getfitla_v3/NutritionListAdapter.java:
```diff
@@ -45,10 +41,10 @@ public void onBindViewHolder(itemViewHolder holder, int position) {
-        holder.textViewRating.setText(String.valueOf(nutrition.getRating()));
-        holder.textViewPrice.setText(String.valueOf(nutrition.getPrice()));
+        //holder.textViewRating.setText(String.valueOf(nutrition.getRating()));
+        //holder.textViewPrice.setText(String.valueOf(nutrition.getPrice()));
```
delete

justinnhli reviewed (changes requested):
In addition to the in-line comments, also clean up the vertical whitespace - lots of places have multiple blank lines that make the code difficult to read.

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/ExerciseItemFormat.java:
```diff
@@ -1,46 +1,97 @@
+
     }
 
     public int getId() {
```
I made this comment on Dan's code, but I'm not sure why you need all these methods. Can't you just get the objects attributes by doing: 
```
int id = exampleObject.id;
```
I guess I don't see the point in writing another method to do the same thing

wchen-oxy replied:
I figured out what word I meant to say when I tried to explain why I want to keep my strings private: encapsulation. Keeping variables private ensures that I don't accidentally override anything.
http://javarevisited.blogspot.com/2012/03/what-is-encapsulation-in-java-and-oops.html

@crzeller11

crzeller11 replied:
Ah ok, I wasn't familiar with this, but this is fine!

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/ExerciseListAdapter.java:
```diff
@@ -39,25 +36,28 @@ public itemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public int getItemCount() {
         return fullList == null ? 0 : fullList.size();
```
This is probably fine, but I usually just write 
```
return fullList.size();
``` 
I don't think you need to catch the exception when it's 0

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,210 @@
     private NutritionListAdapter mAdapter;
-    private Context mContext;
+
+    private void processJson(JSONObject object) {
```
So you can put both the JSON process/object creation together with the code that you actually render the view with, but it would be cleaner to have one class where you read in all this JSON shit and write a method that throws it all in an ArrayList, and then have  separate class to do onCreate

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,210 @@
+                int counter = 0;
+                //added try/catch exceptions because of null values in Json files
+                try {
+                    name = row.getJSONArray("c").getJSONObject(counter).optString("v");
```
You repeat this line a lot, write a method to make it cleaner.

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java:
```diff
@@ -1,34 +1,53 @@
+
+        NutritionItemFormat CurItem = (NutritionItemFormat) data.getParcelable("RecipeInfo");
+        passedid = CurItem.getId();
+        passedname = CurItem.getName();
```
you only need passedname and passedprocess to set up your text view below, you can delete everything else.

wchen-oxy replied:
@Danhuth Can u give me the equivalent NUTRITION_SHEET_URL

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java:
```diff
@@ -1,34 +1,53 @@
 //Please ignore this page for now, this is a dummy page
-public class NutritionDetailActivity extends AppCompatActivity{
+public class NutritionDetailActivity extends AppCompatActivity {
+    private int passedid;
```
Only declare what you need to show in the view. The only variables you need are passedname and passedprocess

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:
```diff
@@ -1,46 +1,130 @@
+        this.ingredients = ingredients;
     }
 
     public int getId() {
```
Again, I think these methods are unnecessary

wchen-oxy committed:
Lol

wchen-oxy committed:
Truncated Code 

Wrote an add function for all the elements of an object for Nutrition Class

wchen-oxy committed:
Updated Layouts

Exercise, Detailed Exercise are changed

wchen-oxy committed:
Adjusted Format for all files

Should comply with Justin/Chloe's requests

wchen-oxy commented:
@justinnhli @crzeller11 

Everything should be fixed now. If there are unused passed variables in the detail activity, that's only because we haven't used it in the layouts yet. Joey/Me will be taking care of as soon as the rest are merged onto the main copy. 

I kept the extra get methods because of encapsulation.

justinnhli commented:
There are still unfixed comments. Scroll back up this page to see Chloe and my notes on your code.

justinnhli commented on app/src/main/res/values/strings.xml:
```diff
@@ -21,4 +21,17 @@
     <string name="Title" translatable="false">Get Fit L.A.</string>
 
-</resources>
+    =======
```
I'm assuming this is left over from a merge conflict resolution

justinnhli replied:
This still needs to be fixed, I think.

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Nutrition.java:
```diff
@@ -1,86 +1,162 @@
+                counter = Integer.parseInt(Temp.get(1).toString());
 
+                Temp = add(x, counter, row);
+                shortdesc = Temp.get(0).toString();
```
I have so many questions about this above section of code. Why is the arraylist named `Temp` (with a capital T, even)? What is `x`? Why do you keep adding `x` to `Temp` then getting the first thing?

wchen-oxy replied:
I use these both in for loop, where each loop contains a counter starting at zero. This was a dumb workaround to return counter (int) and to return the value (string) from the add() method. They way I had written it wasn't letting me return two things and/or modify the counter when the add() was called. I'm not sure how to call the add() method, modify a counter that was initialized in the for loop and return a value found in the Json.

justinnhli replied:
Hm. Can you do something like (untested):

```java
public String getFromRow(JSONObject row, int counter) {
    try {
        //check if valid
        return row.getJSONArray("c").getJSONObject(counter).optString("v");
    } catch (JSONException ex) {
        //exception for null values
        return "";
    }
}
```

Then in `processJson`:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter);
counter++;
String prepTime = getFromRow(row, counter);
counter++;
String servingSize = getFromRow(row, counter);
counter++;
String calories = getFromRow(row, counter);
counter++;
String equipment = getFromRow(row, counter);
counter++;
String ingredients = getFromRow(row, counter);
counter++;
String process = getFromRow(row, counter);
counter++;
String shortdesc = getFromRow(row, counter);
counter++;

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

justinnhli replied:
Or if you want to be fancy, the second part can be:

```java
//initialize secondary counter to control for the index
int counter = 0;

String name = getFromRow(row, counter++);
String prepTime = getFromRow(row, counter++);
String servingSize = getFromRow(row, counter++);
String calories = getFromRow(row, counter++);
String equipment = getFromRow(row, counter++);
String ingredients = getFromRow(row, counter++);
String process = getFromRow(row, counter++);
String shortdesc = getFromRow(row, counter++);

// Temporary fix until we have images
int image = R.drawable.nutrition;
```

The first version is probably easier to understand.

justinnhli replied:
I take everything back. This is the better way:

```java
NutritionItemFormat nutrition = new NutritionItemFormat(
    row_id,
    getFromRow(row, 0), // name
    getFromRow(row, 7), // shortdesc
    image,
    getFromRow(row, 1), // prepTime
    getFromRow(row, 2), // servingSize
    getFromRow(row, 3), // calories
    getFromRow(row, 4), // equipment
    getFromRow(row, 6), // process
    getFromRow(row, 5), // ingredients
    rating,
    price
);
```

crzeller11 replied:
^I agree with Justin's code, it looks much cleaner and it's much easier to see what's going on

wchen-oxy replied:
I'll give it a shot!

crzeller11 commented on app/src/main/java/practicum/getfitla_v3/NutritionDetailActivity.java:
```diff
@@ -1,34 +1,53 @@
+        //unpack the bundled information from previous activity
+        Bundle data = getIntent().getExtras();
+
+        NutritionItemFormat CurItem = (NutritionItemFormat) data.getParcelable("RecipeInfo");
```
This whole section could be made simpler by doing this for each TextView you're setting up:
```
TextView title = (TextView) findViewById(R.id.recipe_title);
title.setText(CurItem.getName());
```

justinnhli replied:
Or even:

```java
( (TextView) findViewById(R.id.recipe_title) ).setText(CurItem.getName());
```

wchen-oxy committed:
Finished All Pull Request Comments (Hopefully)

wchen-oxy committed:
White Space removed

crzeller11 commented:
@justinnhli looks fine to me, but I'll wait for you to look before I merge

justinnhli commented on app/src/main/java/practicum/getfitla_v3/Exercise.java:
```diff
@@ -1,67 +1,57 @@
-                String shortdesc = Jshortdesc.optString("v");
-                System.out.println(shortdesc);
+                //counter for index
+                int id;
```
I think this is unused?

justinnhli commented on app/src/main/java/practicum/getfitla_v3/NutritionItemFormat.java:
```diff
@@ -1,46 +1,117 @@
     private String shortdesc;
-    private double rating;
-    private double price;
+    private String price;
```
What's the reasoning for price being an String?

wchen-oxy replied:
For $

justinnhli replied:
For displaying, sure, but the data itself is a double. Only convert it to a string on display.

wchen-oxy replied:
That and if they accidentally input $ or another symbol then it won't cause an error

crzeller11 replied:
@justinnhli they aren't going to need the number, other than to display it, so is it really necessary that they have it stored as a double?

justinnhli replied:
In general it's better to follow the type of the data. It's less surprising for someone reading it, and there's always the possibility that in the future it will be used in adding, etc.

wchen-oxy committed:
More Last Minute Comments

wchen-oxy commented:
Okay, @justinnhli Fixed and justified the string.
